compiler/Translate.ml

```diff
@@ -195,6 +195,17 @@ let translate_function_to_pure (trans_ctx : trans_ctx)
   | None -> SymbolicToPure.translate_fun_decl ctx None
   | Some (_, ast) -> SymbolicToPure.translate_fun_decl ctx (Some ast)
 
+let translate_function_to_pure (trans_ctx : trans_ctx)
+    (pure_type_decls : Pure.type_decl Pure.TypeDeclId.Map.t)
+    (fun_dsigs : Pure.decomposed_fun_sig FunDeclId.Map.t) (fdef : fun_decl) :
+    pure_fun_translation_no_loops option =
+  try
+    Some
+      (translate_function_to_pure_hook trans_ctx pure_type_decls fun_dsigs fdef)
+  with CFailure (meta, _) ->
+    let () = save_error __FILE__ __LINE__ meta "Could not generate code, see previous error" in
```

Comment:
The comment should be something along the lines of "Could not translate function ... because of previous error"